pkg/services/sqlstore/team.go

```diff
@@ -24,26 +25,54 @@ func init() {
 	bus.AddHandler("sql", IsAdminOfTeams)
 }
 
-func getTeamSearchSqlBase() string {
+func getFilteredUsers(signedInUser *models.SignedInUser, hiddenUsers map[string]struct{}) []string {
+	filteredUsers := make([]string, 0, len(hiddenUsers))
+	if signedInUser == nil || signedInUser.IsGrafanaAdmin {
+		return filteredUsers
+	}
+
+	for u := range hiddenUsers {
+		if u == signedInUser.Login {
+			continue
+		}
+		filteredUsers = append(filteredUsers, u)
+	}
+
+	return filteredUsers
+}
+
+func getTeamMemberCount(filteredUsers []string) string {
+	if len(filteredUsers) > 0 {
+		return `(SELECT COUNT(*) FROM team_member 
+			INNER JOIN user ON team_member.user_id = user.id
+			WHERE team_member.team_id = team.id AND user.login NOT IN (?` +
+			strings.Repeat(",?", len(filteredUsers)-1) + ")" +
+			`) AS member_count `
+	}
+
+	return "(SELECT COUNT(*) FROM team_member WHERE team_member.team_id = team.id) AS member_count "
+}
+
+func getTeamSearchSqlBase(filteredUsers []string) string {
 	return `SELECT
-		team.id as id,
+		team.id AS id,
 		team.org_id,
-		team.name as name,
-		team.email as email,
-		(SELECT COUNT(*) from team_member where team_member.team_id = team.id) as member_count,
-		team_member.permission
-		FROM team as team
-		INNER JOIN team_member on team.id = team_member.team_id AND team_member.user_id = ? `
+		team.name AS name,
+		team.email AS email,
+		team_member.permission, ` +
+		getTeamMemberCount(filteredUsers) +
+		`FROM team AS team
```

Comment:
_very_ much a nitpick, but to avoid problems in the future I think we should have a space here

```suggestion
		` FROM team AS team
```